contracts/poap/src/contract.rs

```diff
@@ -0,0 +1,1086 @@
+        let execute_result = execute(deps.as_mut(), env, info, msg);
+        assert_eq!(Unauthorized {}, execute_result.unwrap_err());
+    }
+
+    #[test]
+    fn disable_mint() {
+        let mut deps = mock_dependencies();
+        let env = mock_env();
+        let info = mock_info(ADMIN, &vec![]);
+
+        do_instantiate(deps.as_mut());
+
+        let msg = ExecuteMsg::DisableMint {};
+        execute(deps.as_mut(), env, info, msg).unwrap();
+
+        let config = CONFIG.load(&deps.storage).unwrap();
+        assert_eq!(false, config.mint_enabled);
+    }
+
+    #[test]
+    fn normal_user_can_not_disable_mint() {
+        let mut deps = mock_dependencies();
+        let env = mock_env();
+        let info = mock_info(USER, &vec![]);
+
+        do_instantiate(deps.as_mut());
+
+        let msg = ExecuteMsg::DisableMint {};
+        let execute_result = execute(deps.as_mut(), env, info, msg);
+        assert_eq!(Unauthorized {}, execute_result.unwrap_err());
+    }
+
+    #[test]
+    fn creator_change_event_info_properly() {
+        let mut deps = mock_dependencies();
+        let mut env = mock_env();
+        let info = mock_info(CREATOR, &vec![]);
+        let new_start_time = Timestamp::from_seconds(env.block.time.seconds() + 100);
+        let new_end_time = Timestamp::from_seconds(env.block.time.seconds() + 400);
+
+        do_instantiate(deps.as_mut());
+
+        env.block.time = Timestamp::from_seconds(INITIAL_BLOCK_TIME_SECONDS);
+
+        let msg = ExecuteMsg::UpdateEventInfo {
+            start_time: new_start_time.clone(),
+            end_time: new_end_time.clone(),
+        };
+
+        execute(deps.as_mut(), env, info, msg).unwrap();
+
+        let event_info = EVENT_INFO.load(&deps.storage).unwrap();
+        assert_eq!(new_start_time, event_info.start_time);
+        assert_eq!(new_end_time, event_info.end_time)
+    }
+
+    #[test]
+    fn non_creator_change_event_info() {
+        let mut deps = mock_dependencies();
+        let env = mock_env();
+        let info = mock_info(USER, &vec![]);
+        let new_start_time = Timestamp::from_seconds(env.block.time.seconds() + 100);
+        let new_end_time = Timestamp::from_seconds(env.block.time.seconds() + 400);
+        let msg = ExecuteMsg::UpdateEventInfo {
+            start_time: new_start_time.clone(),
+            end_time: new_end_time.clone(),
+        };
+
+        do_instantiate(deps.as_mut());
+
+        let result = execute(deps.as_mut(), env, info, msg.clone());
+        // User should not be authorized to update the event info
+        assert_eq!(Unauthorized {}, result.unwrap_err());
+
+        let env = mock_env();
+        let info = mock_info(ADMIN, &vec![]);
+
+        let result = execute(deps.as_mut(), env, info, msg);
+        // Admin should not be authorized to update the event info
+        assert_eq!(Unauthorized {}, result.unwrap_err());
+    }
+
+    #[test]
+    fn event_info_update_only_if_event_not_started_or_ended() {
```

Comment:
Can it be splitted into multiple tests to ensure that one test for a check? say:
1. event_info_update_after_event_started_error
2. event_info_update_after_event_terminated_error